config/routes.rb

```diff
@@ -47,7 +47,8 @@
       resources :topics
     end
     resources :topics do
-      resources :posts
+      resources :posts 
+      resources :flags, only: [:create]
```

Comment:
I think you want to have flags nested under posts instead. You can put it either inside the topics block or under posts on line 53.